esdlvalidator/validation/functions/check_not_null.py

```diff
@@ -26,27 +27,50 @@ def execute(self):
         # Some esdl entity values have a default of undefined or none when not set
         include.extend(["undefined", "none"])
         value = self.value
+        msg = {"offending_asset": self.value.id}
```

Comment:
A check doesn't have to be on a esdl value, for instance this value can also be a result of a select sum. In this case there is no value.id

Reply:
As a side note: In EMF or pyEcore you can check if a certain attribute has been set by using the `eIsSet(feature)` of an `EObject`. E.g. `pipe.eIsSet('name')` would return `True` if a name has been set.